From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH] reiserfs: reset errval after initializing bitmap cache
References: 216956

 Callers after reiserfs_init_bitmap_cache() expect errval to contain -EINVAL
 until much later. If a condition fails before errval is reset later,
 reiserfs_fill_super() will mistakenly return 0, causing an Oops in
 do_add_mount(). This patch resets errval to -EINVAL after the call.

 I view this as a temporary fix and real error codes should be used throughout
 reiserfs_fill_super().

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

--

 fs/reiserfs/super.c |    1 +
 1 file changed, 1 insertion(+)

diff -ruNpX dontdiff linux-2.6.18/fs/reiserfs/super.c linux-2.6.18.devel/fs/reiserfs/super.c
--- linux-2.6.18/fs/reiserfs/super.c	2006-11-01 11:58:01.000000000 -0500
+++ linux-2.6.18.devel/fs/reiserfs/super.c	2006-11-01 12:01:36.000000000 -0500
@@ -1620,6 +1620,7 @@ static int reiserfs_fill_super(struct su
 		      "jmacd-8: reiserfs_fill_super: unable to read bitmap");
 		goto error;
 	}
+	errval = -EINVAL;
 #ifdef CONFIG_REISERFS_CHECK
 	SWARN(silent, s, "CONFIG_REISERFS_CHECK is set ON");
 	SWARN(silent, s, "- it is slow mode for debugging.");
